src/v/kafka/server/group.cc

```diff
@@ -1176,6 +1176,38 @@ void group::fail_offset_commit(
 
 void group::reset_tx_state(model::term_id term) { _term = term; }
 
+ss::future<txn_offset_commit_response>
+group::store_txn_offsets(txn_offset_commit_request&& req) {
+    if (_partition->term() != _term) {
+        co_return txn_offset_commit_response(
+          req, error_code::unknown_server_error);
+    }
+
+    auto r = std::move(req);
+    model::producer_identity pid{
+      .id = r.data.producer_id, .epoch = r.data.producer_epoch};
+
+    auto tx_it = _volatile_txs.find(pid);
+
+    if (tx_it == _volatile_txs.end()) {
+        co_return txn_offset_commit_response(
+          req, error_code::unknown_server_error);
```

Comment:
maybe here from a copy/paste, but wouldn't this be `error_code::transaction_id_not_found` type of error

Reply:
`unknown_server_error` is an umbrella error code, I'll add precise error core when I add a specialized error handler.